From f6185d03b85ac27e2417443b9367204f0fc14877 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Sun, 20 Mar 2016 15:33:50 +0200
Subject: [PATCH 268/538] arm64: dts: marvell: use new clock binding on Armada
 AP806

- Moved clock node of AP-806-Z to DTSI of Z1 chip,
  and update the nodes of the Z1 units to use Z1 clock driver

Change-Id: Ie8750a1f5358770131318e0ebba8919c8b6e05ae
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28363
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Shadi Ammouri <shadi@marvell.com>
---
 .../boot/dts/marvell/armada-ap806-z1-quad.dtsi     | 34 ++++++++++++++++++++++
 arch/arm64/boot/dts/marvell/armada-ap806.dtsi      | 25 ++++++----------
 2 files changed, 43 insertions(+), 16 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap806-z1-quad.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806-z1-quad.dtsi
index 14e5572..0dd4f50 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806-z1-quad.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806-z1-quad.dtsi
@@ -82,6 +82,40 @@
 
 	ap806 {
 		config-space {
+			/* For SPI, I2C, and Serial units override the clock */
+			spi0: spi@510600 {
+				clocks = <&ringclk 2>;
+			};
+
+			i2c0: i2c@511000 {
+				clocks = <&ringclk 2>;
+			};
+
+			serial@512000 {
+				clocks = <&ringclk 2>;
+			};
+
+			serial@512100 {
+				clocks = <&ringclk 2>;
+
+			};
+			/* Add clock node for AP-806-Z SoC
+			** The Z1 revision use different driver than A0 revision
+			*/
+			coreclk: clk@0x6F8204 {
+				compatible = "marvell,armada-ap806-core-clock";
+				reg = <0x6F8204 0x04>;
+				#clock-cells = <1>;
+				clock-output-names = "ddr", "ring", "cpu";
+			};
+
+			ringclk: clk@0x6F8250 {
+				compatible = "marvell,armada-ap806-ring-clock";
+				reg = <0x6F8250 0x04>;
+				#clock-cells = <1>;
+				clock-output-names = "ring-0", "ring-2", "ring-3", "ring-4", "ring-5";
+				clocks = <&coreclk 1>;
+			};
 			xor0@400000 {
 				compatible = "marvell,mv-xor-v2-z1";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
index d4174fb..ef04a03 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap806.dtsi
@@ -170,7 +170,7 @@
 				#size-cells = <0>;
 				cell-index = <0>;
 				interrupts = <GIC_SPI 21 IRQ_TYPE_LEVEL_HIGH>;
-				clocks = <&ringclk 2>;
+				clocks = <&syscon 3>;
 				status = "disabled";
 			};
 
@@ -181,7 +181,7 @@
 				#size-cells = <0>;
 				interrupts = <GIC_SPI 20 IRQ_TYPE_LEVEL_HIGH>;
 				timeout-ms = <1000>;
-				clocks = <&ringclk 2>;
+				clocks = <&syscon 3>;
 				status = "disabled";
 			};
 
@@ -191,7 +191,7 @@
 				reg-shift = <2>;
 				interrupts = <GIC_SPI 19 IRQ_TYPE_LEVEL_HIGH>;
 				reg-io-width = <1>;
-				clocks = <&ringclk 2>;
+				clocks = <&syscon 3>;
 				status = "disabled";
 			};
 
@@ -201,7 +201,7 @@
 				reg-shift = <2>;
 				interrupts = <GIC_SPI 29 IRQ_TYPE_LEVEL_HIGH>;
 				reg-io-width = <1>;
-				clocks = <&ringclk 2>;
+				clocks = <&syscon 3>;
 				status = "disabled";
 
 			};
@@ -217,19 +217,12 @@
 				status = "disabled";
 			};
 
-			coreclk: clk@0x6F8204 {
-				compatible = "marvell,armada-ap806-core-clock";
-				reg = <0x6F8204 0x04>;
+			syscon: system-controller@6f4000 {
+				compatible = "marvell,ap806-system-controller", "syscon";
 				#clock-cells = <1>;
-				clock-output-names = "ddr", "ring", "cpu";
-			};
-
-			ringclk: clk@0x6F8250 {
-				compatible = "marvell,armada-ap806-ring-clock";
-				reg = <0x6F8250 0x04>;
-				#clock-cells = <1>;
-				clock-output-names = "ring-0", "ring-2", "ring-3", "ring-4", "ring-5";
-				clocks = <&coreclk 1>;
+				clock-output-names = "ap-cpu-cluster-0", "ap-cpu-cluster-1",
+						     "ap-fixed", "ap-mss";
+				reg = <0x6f4000 0x1000>;
 			};
 
 			xor0@400000 {
-- 
1.9.1

